### ISSUE TITLE
Possible test flakyness: `test_logging.TimedRotatingFileHandlerTest.test_rollover_at_midnight` failure

### ISSUE THREAD

**Issue by sobolevn** [python/cpython#116455]
# Bug report

This failed on unrelated change:

```
 ======================================================================
FAIL: test_rollover_at_midnight (test.test_logging.TimedRotatingFileHandlerTest.test_rollover_at_midnight)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/test/test_logging.py", line 6111, in test_rollover_at_midnight
    self.assertIn(f'testing2 {i}', line)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 'testing2 0' not found in '2024-03-07 08:26:34,002 testing1 0\n'
```

Link: https://github.com/python/cpython/actions/runs/8185043132/job/22380600561

**Comment by serhiy-storchaka** [comment]
Should be fixed by #116409.

**Comment by erlend-aasland** [comment]
Flakiness observed in https://github.com/python/cpython/actions/runs/8221464633/job/22481869409
